lib/metrics.rb

```diff
@@ -1,29 +1,51 @@
 class Metrics
-  def self.with_object(object)
-    prefix = object.class.name.underscore.tr("/", ".")
-    new(prefix)
+  include Singleton
+
+  def gauge(event, count, labels = {}, description = nil)
+    record_metric(:gauge, event, count, labels, description)
+  end
+
+  def increment(event, labels = {}, description = nil)
+    record_metric(:counter, event, 1, labels, description)
+  end
```

Comment:
**suggestion** (non-blocking): Calling this something like `inc_counter` might be nicer because it lets the caller understand what they are working with. Some libraries tend to allow inc on gauges and other metric types.

So the interface will always make clear what we are working with this way;
```ruby
Metrics.gauge(...)
Metrics.histogram(...)
Metrics.counter_inc(...)   # or maybe just Metrics.counter(...)?
```

Reply:
This should be fine as is, technically yes, you can increment a gauge also, but I wanted to be explicit that only counters are to be incremented